examples/example.rs

```diff
@@ -43,7 +45,9 @@ fn main() {
     }
 
     // middleware is optional and can be registered with `utilize`
-    server.utilize(logger);
+    // issue #15444
+    let logger_handler: fn(&Request, &mut Response) -> MiddlewareResult = logger;
+    server.utilize(logger_handler);
```

Comment:
`server.utilize(logger as fn(&Request, &mut Response) -> MiddlewareResult);` is the common form of this last I saw, but it's still horrible :(